src/main/java/com/lgtm/easymoney/services/impl/RequestServiceImpl.java

```diff
@@ -96,136 +93,67 @@ public Transaction createRequest(User reqBy, User reqTo, BigDecimal amount, Stri
   }
 
   @Override
-  public ResponseEntity<RequestRsp> createRequest(RequestReq req) {
-    // get params
-    Long fromUid = req.getFromUid();      // from = req by
-    Long toUid = req.getToUid();          // to = req to
-    BigDecimal amount = req.getAmount();
-    Category category = req.getCategory();
-    String desc = req.getDescription();
+  public ResourceCreatedRsp createRequest(RequestReq req) {
     // create a request
-    User fromUser = userService.getUserById(fromUid);
-    User toUser = userService.getUserById(toUid);
-    Transaction trans = createRequest(fromUser, toUser, amount, desc, category);
-    // payload
-    // todo need to refactor some of these
-    RequestRsp res = new RequestRsp();
-    res.setSuccess(trans != null);
-    res.setCurrBalance(fromUser.getBalance());
-    res.setRequests(transactionService.generateListResponseFromTransactions(
-            new ArrayList<>(Arrays.asList(trans))));
-    res.setMessage("Request has been created successfully!");
-    if (trans != null) {
-      return ResponseEntity.status(HttpStatus.OK).body(res);
-    }
-    // TODO refactor this?
-    return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(res);
+    Transaction trans = createRequest(
+            userService.getUserById(req.getFromUid()),
+            userService.getUserById(req.getToUid()),
+            req.getAmount(),
+            req.getDescription(),
+            req.getCategory());
+    // response
+    return trans == null ? new ResourceCreatedRsp(null) : new ResourceCreatedRsp(trans.getId());
   }
 
   @Override
-  public boolean validateRequest(Transaction request) {
-    // TODO maybe it's better to handle exceptions here?
-    // TODO validate user?
-    // a valid request = valid users, transaction
-    // status should be pending, otherwise it's
-    // not a valid REQUEST, it's a TRANSFER
-    return existsRequestById(request.getId());
+  public boolean canAcceptDeclineRequest(Long tid, Long fuid, Long tuid) {
+    return
+            getRequestById(tid).getFrom().getId().equals(fuid)
+            && getRequestById(tid).getTo().getId().equals(tuid)
+            && getRequestById(tid).getStatus() == TransactionStatus.TRANS_PENDING;
```

Comment:
Better to assign getRequestById(tid) to a local variable to avoid duplicate DB queries